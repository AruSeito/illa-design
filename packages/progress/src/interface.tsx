import { HTMLAttributes } from "react"

export type ProgressColorScheme =
  string
  | "white"
  | "blackAlpha"
  | "gray"
  | "grayBlue"
  | "red"
  | "orange"
  | "yellow"
  | "green"
  | "blue"
  | "cyan"
  | "purple"

export interface ProgressProps extends HTMLAttributes<HTMLDivElement> {
}
