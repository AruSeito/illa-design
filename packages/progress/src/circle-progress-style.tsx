import { css } from "@emotion/css"
import { globalColor, illaPrefix } from "@illa-design/theme"

export const applyCircleStatus = css`
  position: absolute;
  display: inline-flex;
  font-size: 16px;
`

export const applyProgressText = css`
  position: absolute;
  line-height: 22px;
  font-size: 14px;
  color: ${globalColor(`--${illaPrefix}-gray-04`)};
`
