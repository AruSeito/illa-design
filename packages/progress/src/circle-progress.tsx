/** @jsxImportSource @emotion/react */
import { forwardRef, ReactNode } from "react"
import { ProgressProps } from "./interface"
import {
  applyCircleContainer,
  applyCircleStatus,
  applyContainer,
  applyProgressContainer,
  applyProgressContainerBg,
  applyProgressText,
} from "./circle-progress-style"
import { ErrorIcon, SuccessIcon } from "@illa-design/icon"

export const CircleProgress = forwardRef<HTMLDivElement, ProgressProps>((props, ref) => {

  const {
    type = "line",
    steps,
    status = "normal",
    color = "blue",
    width = "80px",
    strokeWidth = "4px",
    trailColor = "gray",
    showText = true,
    formatText = (percent: number) => {
      return `${percent}%`
    },
    percent = 0,
    ...otherProps
  } = props

  const radius = `calc((${width} - ${strokeWidth}) / 2)`
  const center = `calc(${width} / 2)`

  let finalColor: string

  switch (status) {
    case "normal":
      finalColor = color
      break
    case "success":
      finalColor = "green"
      break
    case "error":
      finalColor = "red"
      break
  }

  let statusComponent: ReactNode
  switch (status) {
    case "success":
      statusComponent = <SuccessIcon />
      break
    case "error":
      statusComponent = <ErrorIcon />
      break
  }

  return <div css={applyContainer(width)} ref={ref} {...otherProps}>
    <svg css={applyCircleContainer(width)}>
      <circle
        css={applyProgressContainerBg(trailColor, width, strokeWidth)}
        fill="none"
        cx={center}
        cy={center}
        r={radius}
        strokeWidth={strokeWidth} />
      <circle
        css={applyProgressContainer(finalColor, width, strokeWidth, percent)}
        fill="none"
        cx={center}
        cy={center}
        r={radius}
        strokeWidth={strokeWidth}
        strokeLinecap="round" />
    </svg>
    {status == "normal" && showText && <span css={applyProgressText}>
      {formatText(percent)}
    </span>}
    {status != "normal" && <span css={applyCircleStatus}>{statusComponent}</span>}
  </div>
})

CircleProgress.displayName = "CircleProgress"