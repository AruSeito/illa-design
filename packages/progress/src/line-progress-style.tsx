import { css } from "@emotion/react"
import { SerializedStyles } from "@storybook/theming"
import { ProgressColorScheme } from "./interface"
import { globalColor, illaPrefix } from "@illa-design/theme"

const colorSchemes: ProgressColorScheme[] = ["white", "blackAlpha", "gray", "grayBlue", "red", "orange", "yellow", "green", "blue", "cyan", "purple"]

export const applyProgressContainer = css`
  vertical-align: middle;
  align-items: center;
  display: inline-flex;
  flex-direction: row;
`

export const applyProgressText = css`
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: ${globalColor(`--${illaPrefix}-gray-04`)};
`

export const applyStatusIcon = css`
  display: inline-flex;
  font-size: 12px;
  margin-left: 8px;
`

export function applyLineContainer(width: string, stokeWidth: string): SerializedStyles {
  return css`
    display: inline-block;
    height: ${stokeWidth};
    width: ${width};
    position: relative;
  `
}

export function applySpace(): SerializedStyles {
  return css`
    position: absolute;
  `
}

export function applyLineProgress(percent: number, stokeWidth: string, colorScheme: ProgressColorScheme): SerializedStyles {
  let color
  if (colorScheme == "gray") {
    color = globalColor(`--${illaPrefix}-${colorScheme}-02`)
  } else {
    color = colorSchemes.includes(colorScheme) ? globalColor(`--${illaPrefix}-${colorScheme}-03`) : colorScheme
  }
  return css`
    position: absolute;
    width: ${percent}%;
    border-radius: calc(${stokeWidth} / 2);
    background-color: ${color};
    height: ${stokeWidth};
  `
}

export function applyLineProgressBg(stokeWidth: string, trailColorScheme: ProgressColorScheme): SerializedStyles {
  let color
  if (trailColorScheme == "gray") {
    color = globalColor(`--${illaPrefix}-${trailColorScheme}-08`)
  } else {
    color = colorSchemes.includes(trailColorScheme) ? globalColor(`--${illaPrefix}-${trailColorScheme}-06`) : trailColorScheme
  }
  return css`
    position: absolute;
    background-color: ${color};
    width: 100%;
    border-radius: calc(${stokeWidth} / 2);
    height: ${stokeWidth};
  `
}