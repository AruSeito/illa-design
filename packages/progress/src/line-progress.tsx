import { forwardRef, ReactNode } from "react"
import { ProgressProps } from "./interface"
import {
  applyLineContainer,
  applyLineProgress,
  applyLineProgressBg,
  applyLineProgressBgStep,
  applyLineProgressStep,
  applyProgressContainer,
  applyProgressText,
  applySpace,
  applyStatusIcon,
} from "./line-progress-style"
import { SuccessIcon, WarningCircleIcon } from "@illa-design/icon"
import { Space } from "@illa-design/space"
import { cx, globalColor, illaPrefix } from "@illa-design/theme"

export const LineProgress = forwardRef<HTMLDivElement, ProgressProps>(
  (props, ref) => {
    const {
      steps = 1,
      status = "normal",
      color = "blue",
      trailColor = "gray",
      className,
      showText = true,
      formatText = (percent: number) => {
        return `${percent}%`
      },
      percent = 0,
      strokeWidth = "4px",
      width = "320px",
      ...otherProps
    } = props

    let finalColor: string

    switch (status) {
      case "normal":
        finalColor = color
        break
      case "success":
        finalColor = "green"
        break
      case "error":
        finalColor = "red"
        break
    }

    let statusComponent: ReactNode
    switch (status) {
      case "success":
        statusComponent = (
          <SuccessIcon color={globalColor(`--${illaPrefix}-green-03`)} />
        )
        break
      case "error":
        statusComponent = (
          <WarningCircleIcon color={globalColor(`--${illaPrefix}-red-03`)} />
        )
        break
    }

    if (steps == 1) {
      return (
        <div
          ref={ref}
          className={cx(applyProgressContainer, className)}
          {...otherProps}
        >
          <div className={applyLineContainer(width, strokeWidth)}>
            <div className={applyLineProgressBg(strokeWidth, trailColor)} />
            <div
              className={applyLineProgress(percent, strokeWidth, finalColor)}
            />
          </div>
          {showText && (
            <span className={applyProgressText}>{formatText(percent)}</span>
          )}
          {status != "normal" && (
            <span className={applyStatusIcon}>{statusComponent}</span>
          )}
        </div>
      )
    } else {
      let lineProgressContainer = []
      let lineProgressBgContainer = []

      for (let i = 0; i < steps; i++) {
        lineProgressContainer.push(
          <div
            key={i}
            className={applyLineProgressStep(
              percent,
              strokeWidth,
              `calc((${width} - (${steps} - 1) * 4px) / ${steps})`,
              finalColor,
              steps,
              i,
            )}
          />,
        )
        lineProgressBgContainer.push(
          <div
            key={i}
            className={applyLineProgressBgStep(
              strokeWidth,
              `calc((${width} - (${steps} - 1) * 4px) / ${steps})`,
              trailColor,
            )}
          />,
        )
      }

      return (
        <div ref={ref} className={applyProgressContainer} {...otherProps}>
          <div className={applyLineContainer(width, strokeWidth)}>
            <Space className={applySpace()} size="4px">
              {lineProgressBgContainer}
            </Space>
            <Space className={applySpace()} size="4px">
              {lineProgressContainer}
            </Space>
          </div>
          {showText && (
            <span className={applyProgressText}>{formatText(percent)}</span>
          )}
          {status != "normal" && (
            <span className={applyStatusIcon}>{statusComponent}</span>
          )}
        </div>
      )
    }
  },
)

LineProgress.displayName = "LineProgress"
