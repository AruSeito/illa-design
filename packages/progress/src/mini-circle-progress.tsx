import { forwardRef, ReactNode } from "react"
import { ProgressProps } from "./interface"
import { SuccessIcon, WarningCircleIcon } from "@illa-design/icon"
import { Trigger } from "@illa-design/trigger"
import { applyContainer } from "./common-style"
import {
  applyCircleProgressContainer,
  applyCircleStatus,
  applyCircleSvgContainer,
} from "./mini-circle-progress-style"
import { globalColor, illaPrefix } from "@illa-design/theme"
import { cx } from "@emotion/css"

export const MiniCircleProgress = forwardRef<HTMLDivElement, ProgressProps>(
  (props, ref) => {
    const {
      type = "line",
      status = "normal",
      color = "blue",
      width = "16px",
      strokeWidth = "4px",
      trailColor = "gray",
      className,
      showText = true,
      formatText = (percent: number) => {
        return `${percent}%`
      },
      percent = 0,
      steps,
      ...otherProps
    } = props

    const radius = `calc(${width} / 2)`
    const center = `calc(${width} / 2)`

    let finalColor: string

    switch (status) {
      case "normal":
        finalColor = color
        break
      case "success":
        finalColor = "green"
        break
      case "error":
        finalColor = "red"
        break
    }

    let statusComponent: ReactNode
    switch (status) {
      case "success":
        statusComponent = (
          <SuccessIcon color={globalColor(`--${illaPrefix}-green-03`)} />
        )
        break
      case "error":
        statusComponent = (
          <WarningCircleIcon color={globalColor(`--${illaPrefix}-red-03`)} />
        )
        break
    }

    return (
      <div
        className={cx(applyContainer(width), className)}
        ref={ref}
        {...otherProps}
      >
        <Trigger disabled={!showText} content={formatText(percent)}>
          {(status == "normal" || status == "error") && (
            <svg className={applyCircleSvgContainer(width, trailColor)}>
              <circle
                className={applyCircleProgressContainer(
                  finalColor,
                  width,
                  percent,
                )}
                fill="none"
                cx={center}
                cy={center}
                r={radius}
                strokeWidth={width}
              />
            </svg>
          )}
          {status == "success" && (
            <span className={applyCircleStatus}>{statusComponent}</span>
          )}
        </Trigger>
      </div>
    )
  },
)

MiniCircleProgress.displayName = "MiniCircleProgress"
