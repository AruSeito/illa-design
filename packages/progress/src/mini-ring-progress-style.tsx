import { css } from "@emotion/react"

export const applyRingStatus = css`
  display: inline-flex;
  font-size: 16px;
`
