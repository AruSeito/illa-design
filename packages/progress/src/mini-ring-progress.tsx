import { forwardRef } from "react"
import { ProgressProps } from "./interface"
import { Trigger } from "@illa-design/trigger"
import {
  applyContainer,
  applyProgressContainer,
  applyProgressContainerBg,
  applySvgContainer,
} from "./common-style"
import { cx } from "@emotion/css"

export const MiniRingProgress = forwardRef<HTMLDivElement, ProgressProps>(
  (props, ref) => {
    const {
      type = "line",
      steps,
      status = "normal",
      color = "blue",
      className,
      width = "16px",
      strokeWidth = "4px",
      trailColor = "gray",
      showText = true,
      formatText = (percent: number) => {
        return `${percent}%`
      },
      percent = 0,
      ...otherProps
    } = props

    const radius = `calc((${width} - ${strokeWidth}) / 2)`
    const center = `calc(${width} / 2)`

    let finalColor: string

    switch (status) {
      case "normal":
        finalColor = color
        break
      case "success":
        finalColor = "green"
        break
      case "error":
        finalColor = "red"
        break
    }

    return (
      <div
        className={cx(applyContainer(width), className)}
        ref={ref}
        {...otherProps}
      >
        <Trigger disabled={!showText} content={formatText(percent)}>
          <svg className={applySvgContainer(width)}>
            <circle
              className={applyProgressContainerBg(
                trailColor,
                width,
                strokeWidth,
              )}
              fill="none"
              cx={center}
              cy={center}
              r={radius}
              strokeWidth={strokeWidth}
            />
            <circle
              className={applyProgressContainer(
                finalColor,
                width,
                strokeWidth,
                percent,
              )}
              fill="none"
              cx={center}
              cy={center}
              r={radius}
              strokeWidth={strokeWidth}
            />
          </svg>
        </Trigger>
      </div>
    )
  },
)

MiniRingProgress.displayName = "MiniRingProgress"
