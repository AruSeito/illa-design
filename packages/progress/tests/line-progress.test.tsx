import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { Progress } from "../src"

test("LineProgress renders correctly", () => {
  render(
    <Progress data-testid="test-line-progress" type="line" />,
  )
  expect(screen.getByTestId("test-line-progress")).toBeInTheDocument()
})

test("LineProgress renders with colors", () => {
  render(
    <div>
      <Progress data-testid="test-line-progress-red" type="line" color="red" trailColor="red" />,
      <Progress data-testid="test-line-progress-gray" type="line" color="gray" trailColor="gray" />,
      <Progress data-testid="test-line-progress-custom" type="line" color="#123123" trailColor="#321321" />,
    </div>,
  )
  expect(screen.getByTestId("test-line-progress-red")).toMatchSnapshot()
  expect(screen.getByTestId("test-line-progress-gray")).toMatchSnapshot()
  expect(screen.getByTestId("test-line-progress-custom")).toMatchSnapshot()
})

test("LineProgress renders with different status", () => {
  render(
    <div>
      <Progress status="success" />,
      <Progress status="error" />,
    </div>,
  )
  expect(screen.getByTitle("SuccessIcon")).toBeInTheDocument()
  expect(screen.getByTitle("ErrorIcon")).toBeInTheDocument()
})

test("LineProgress renders without text", () => {
  render(<Progress percent={50} showText={false} />)
  expect(screen.queryByText("50")).not.toBeTruthy()
})

test("LineProgress renders with format text function", () => {
  render(<Progress percent={50} formatText={(percent) => {
    return `test${percent}`
  }
  } />)
  expect(screen.getByText("test50")).toBeInTheDocument()
})

test("LineProgress renders with custom width", () => {
  render(<Progress data-testid="test-line-progress" percent={50} width="100px" strokeWidth="10px" />)
  expect(screen.getByTestId("test-line-progress")).toMatchSnapshot()
})

test("LineProgress renders with steps", () => {
  render(<Progress data-testid="test-line-progress" steps={3} />)
  expect(screen.getByTestId("test-line-progress")).toMatchSnapshot()
})