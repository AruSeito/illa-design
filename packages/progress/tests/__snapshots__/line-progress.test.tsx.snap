// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders with colors 1`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-3 {
  position: absolute;
  background-color: #fcc;
  width: 100%;
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-4 {
  -webkit-transition: width 200ms;
  transition: width 200ms;
  position: absolute;
  width: 0%;
  border-radius: calc(4px / 2);
  background-color: #ff4747;
  height: 4px;
}

.emotion-5 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0 emotion-0"
  data-testid="test-progress-red"
>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-3"
    />
    <div
      class="emotion-4"
    />
  </div>
  <span
    class="emotion-5"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with colors 2`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-3 {
  position: absolute;
  background-color: #ebebeb;
  width: 100%;
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-4 {
  -webkit-transition: width 200ms;
  transition: width 200ms;
  position: absolute;
  width: 0%;
  border-radius: calc(4px / 2);
  background-color: #1f1f1f;
  height: 4px;
}

.emotion-5 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0 emotion-0"
  data-testid="test-progress-gray"
>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-3"
    />
    <div
      class="emotion-4"
    />
  </div>
  <span
    class="emotion-5"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with colors 3`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-3 {
  position: absolute;
  background-color: #321321;
  width: 100%;
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-4 {
  -webkit-transition: width 200ms;
  transition: width 200ms;
  position: absolute;
  width: 0%;
  border-radius: calc(4px / 2);
  background-color: #123123;
  height: 4px;
}

.emotion-5 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0 emotion-0"
  data-testid="test-progress-custom"
>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-3"
    />
    <div
      class="emotion-4"
    />
  </div>
  <span
    class="emotion-5"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with colors 4`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-2 {
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-3 {
  width: 100%;
  position: absolute;
}

.emotion-5 {
  height: 100%;
  display: inline-flex;
  margin-right: 4px;
  margin-bottom: 0px;
}

.emotion-6 {
  background-color: #fcc;
  width: calc((320px - (3 - 1) * 4px) / 3);
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-9 {
  height: 100%;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-15 {
  width: 0px;
  border-radius: calc(4px / 2);
  background-color: #ff4747;
  height: 4px;
}

.emotion-20 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0"
  data-testid="test-progress-red-step"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-6"
        />
      </div>
    </div>
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-15"
        />
      </div>
    </div>
  </div>
  <span
    class="emotion-20"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with colors 5`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-2 {
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-3 {
  width: 100%;
  position: absolute;
}

.emotion-5 {
  height: 100%;
  display: inline-flex;
  margin-right: 4px;
  margin-bottom: 0px;
}

.emotion-6 {
  background-color: #ebebeb;
  width: calc((320px - (3 - 1) * 4px) / 3);
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-9 {
  height: 100%;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-15 {
  width: 0px;
  border-radius: calc(4px / 2);
  background-color: #1f1f1f;
  height: 4px;
}

.emotion-20 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0"
  data-testid="test-progress-gray-step"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-6"
        />
      </div>
    </div>
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-15"
        />
      </div>
    </div>
  </div>
  <span
    class="emotion-20"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with colors 6`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-2 {
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-3 {
  width: 100%;
  position: absolute;
}

.emotion-5 {
  height: 100%;
  display: inline-flex;
  margin-right: 4px;
  margin-bottom: 0px;
}

.emotion-6 {
  background-color: #321321;
  width: calc((320px - (3 - 1) * 4px) / 3);
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-9 {
  height: 100%;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-15 {
  width: 0px;
  border-radius: calc(4px / 2);
  background-color: #123123;
  height: 4px;
}

.emotion-20 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0"
  data-testid="test-progress-custom-step"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-6"
        />
      </div>
    </div>
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-15"
        />
      </div>
    </div>
  </div>
  <span
    class="emotion-20"
  >
    0%
  </span>
</div>
`;

exports[`Progress renders with custom width 1`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: inline-block;
  height: 10px;
  width: 100px;
  position: relative;
}

.emotion-3 {
  position: absolute;
  background-color: #ebebeb;
  width: 100%;
  border-radius: calc(10px / 2);
  height: 10px;
}

.emotion-4 {
  -webkit-transition: width 200ms;
  transition: width 200ms;
  position: absolute;
  width: 50%;
  border-radius: calc(10px / 2);
  background-color: #1e6fff;
  height: 10px;
}

.emotion-5 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0 emotion-0"
  data-testid="test-progress"
>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-3"
    />
    <div
      class="emotion-4"
    />
  </div>
  <span
    class="emotion-5"
  >
    50%
  </span>
</div>
`;

exports[`Progress renders with steps 1`] = `
.emotion-0 {
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  display: inline-block;
  height: 4px;
  width: 320px;
  position: relative;
}

.emotion-2 {
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-3 {
  width: 100%;
  position: absolute;
}

.emotion-5 {
  height: 100%;
  display: inline-flex;
  margin-right: 4px;
  margin-bottom: 0px;
}

.emotion-6 {
  background-color: #ebebeb;
  width: calc((320px - (3 - 1) * 4px) / 3);
  border-radius: calc(4px / 2);
  height: 4px;
}

.emotion-9 {
  height: 100%;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

.emotion-15 {
  width: calc((320px - (3 - 1) * 4px) / 3);
  border-radius: calc(4px / 2);
  background-color: #1e6fff;
  height: 4px;
}

.emotion-17 {
  width: 0px;
  border-radius: calc(4px / 2);
  background-color: #1e6fff;
  height: 4px;
}

.emotion-20 {
  margin-left: 8px;
  line-height: 14px;
  font-size: 14px;
  color: #999;
}

<div
  class="emotion-0"
  data-testid="test-progress"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-6"
        />
      </div>
    </div>
    <div
      class="emotion-2 emotion-3 emotion-2"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-15"
        />
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-17"
        />
      </div>
      <div
        class="emotion-9"
      >
        <div
          class="emotion-17"
        />
      </div>
    </div>
  </div>
  <span
    class="emotion-20"
  >
    40%
  </span>
</div>
`;
